WatchFacePush/buildSrc/src/main/kotlin/TokenGenerationTask.gradle.kts

```diff
@@ -57,41 +52,24 @@ abstract class TokenGenerationTask : DefaultTask() {
                 ?: throw GradleException("Cannot load APKs")
         if (artifacts.elements.size != 1)
             throw GradleException("Expected only one APK!")
-        val apkPath = File(artifacts.elements.single().outputFile).toPath()
+        val apk = File(artifacts.elements.single().outputFile)
         val appPackageName = packageName.get()
 
-        val stdOut = ByteArrayOutputStream()
-        val stdErr = ByteArrayOutputStream()
-
-        execOperations.javaexec {
-            classpath = cliToolClasspath.get()
-            mainClass = "com.google.android.wearable.watchface.validator.cli.DwfValidation"
-            args("--apk_path=$apkPath")
-            args("--package_name=$appPackageName")
-
-            standardOutput = stdOut
-            errorOutput = stdErr
-
-            isIgnoreExitValue = true
-        }
-
-        val outputAsText = stdOut.toString()
-        val errorAsText = stdErr.toString()
+        val validator = DwfValidatorFactory.create()
+        val result = validator.validate(apk, appPackageName)
 
-        if (outputAsText.contains("Failed check")) {
-            println(outputAsText)
-            if (errorAsText.isNotEmpty()) {
-                println(errorAsText)
+        val failures = result.failures()
+        if (failures.isNotEmpty()) {
+            failures.forEach { failure ->
+                println("Validation failure: ${failure.name()}: ${failure.category()} - ${failure.failureMessage()}")
             }
             throw GradleException("Watch face validation failed")
```

Comment:
What about adding each one as a supressed exception? Will help with tracking, e.g. if presented in Android Studio?